### PR TITLE
React Testing Library Example

### PR DESCRIPTION
**Pull request by jbolda**
## Motivation

Adding an example with jsdom based tests through jest and the react flavor of testing library.

(Originally merged into a branch from the fork as part of #4)

### CODE REVIEW

File: tests-rtl/src/store.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export function setupStore({ initialState = {} }) {
+  const store = configureStore({
+    initialState: {
+      ...schemaInitialState,
+      ...initialState,
+    },
+  });
+
+  const tsks = [];
```
Comment (neurosnap) [inline tests-rtl/src/store.js]: This needs `api.bootup`

File: tests-rtl/src/store.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  store.run(function* () {
+    const group = yield* parallel(tsks);
+    yield* group;
+  });
```
Comment (neurosnap) [inline tests-rtl/src/store.js]:
```suggestion
  store.run(api.bootup);
```

**Review by neurosnap — CHANGES_REQUESTED** [review]

File: tests-rtl/src/store.js (hunk 2 above)
Comment (jbolda) [reply tests-rtl/src/store.js]: Ahh! Missed that in my copy and pasting. I wonder if we could somehow through a dev only error there, it hasn't been the first time and likely not the last that that is missed.

Comment (neurosnap) [reply tests-rtl/src/store.js]: Hmm yes I think you are on the right track.  We want the user to "do the right thing" and building some error handling like "no tasks were activated" could be useful.

File: tests-rtl/src/store.js (hunk 1 above)
Comment (jbolda) [reply tests-rtl/src/store.js]: Resolved in ffadf58cd9f4c2f30480e8bff8af2db8ce8ad259.

**Review by neurosnap — APPROVED** [review]
great job figuring this all out!  I like the idea of having a tests folder inside our examples, it promotes this repo to be more functional and hopefully will prevent drift from `starfx` core.

**Pull request merged**